command/filemanager.go

```diff
@@ -53,3 +60,133 @@ func (fm *FileManager) CopyInlineFile(name string, fileContent pulumi.StringInpu
 			Sudo:   useSudo,
 		}, opts...)
 }
+
+// CopyRelativeFolder copies recursively a relative folder to a remote folder.
+// The path of the folder is relative to the caller of this function.
+// For example, if this function is called from ~/foo/test.go with remoteFolder="bar"
+// then the full path of the folder will be ~/foo/bar“.
+// This function returns the resources that can be used with `pulumi.DependsOn`.
+func (fm *FileManager) CopyRelativeFolder(relativeFolder string, remoteFolder string, opts ...pulumi.ResourceOption) ([]pulumi.Resource, error) {
+	fullPath, rootFolder, err := getFullPath(relativeFolder, 2)
+	if err != nil {
+		return nil, err
+	}
+
+	return fm.CopyFSFolder(fullPath, os.DirFS(rootFolder), relativeFolder, remoteFolder, opts...)
+}
+
+// CopyRelativeFile copies relative path to a remote path.
+// The relative path is defined in the same way as for `CopyRelativeFolder`.
+// This function returns the resource that can be used with `pulumi.DependsOn`.
+func (fm *FileManager) CopyRelativeFile(relativePath string, remotePath string, opts ...pulumi.ResourceOption) (pulumi.Resource, error) {
+	fullPath, _, err := getFullPath(relativePath, 2)
+	if err != nil {
+		return nil, err
+	}
+
+	return fm.CopyFile(fullPath, remotePath, opts...)
+}
+
+// CopyFSFolder copies recursively a local folder to a remote folder.
+// You may consider using `CopyRelativeFolder` which has a simpler API.
+func (fm *FileManager) CopyFSFolder(
+	resourceName string,
+	folderFS fs.FS,
+	folderPath string,
+	remoteFolder string,
+	opts ...pulumi.ResourceOption) ([]pulumi.Resource, error) {
+	useSudo := true
+	folderCommand, err := fm.CreateDirectory(resourceName, pulumi.String(remoteFolder), useSudo, opts...)
+	if err != nil {
+		return nil, fmt.Errorf("cannot create a temporary folder: %v for resource %v", err, resourceName)
+	}
+
+	files, folders, err := readFilesAndFolders(folderFS, folderPath)
+	if err != nil {
+		return nil, fmt.Errorf("cannot read files and folders for %v", folderPath)
+	}
+
+	var folderResources []pulumi.Resource
+	for _, folder := range folders {
+		destFolder, err := getDestinationPath(folder, folderPath)
+		if err != nil {
+			return nil, err
+		}
+		remotePath := path.Join(remoteFolder, destFolder)
+		resources, err := fm.CreateDirectory("createFolder-"+remotePath, pulumi.String(remotePath), useSudo, pulumi.DependsOn([]pulumi.Resource{folderCommand}))
+		if err != nil {
+			return nil, err
+		}
+		folderResources = append(folderResources, resources)
+	}
+
+	fileResources := []pulumi.Resource{}
+	for _, file := range files {
+		destFile, err := getDestinationPath(file, folderPath)
+
+		if err != nil {
+			return nil, err
+		}
+
+		fileContent, err := fs.ReadFile(folderFS, file)
+		if err != nil {
+			return nil, err
+		}
+		fileCommand, err := fm.CopyInlineFile(
```

Comment:
It's probably better to use the file-system copy file instead of inline (stdin) as it should be faster.

Reply:
Code updated.

Reply:
Push probably missing as code did not change

Reply:
Code reverted to have the same code for embeded and absolute / relative paths.